tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -4013,6 +4014,51 @@ int BPF_KPROBE(trace_security_kernel_read_file)
     return 0;
 }
 
+SEC("kprobe/security_inode_mknod")
+int BPF_KPROBE(trace_security_inode_mknod)
+{
+    if (!should_trace()) {
+        return 0;
+    }
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL) {
+        return 0;
+    }
+
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    context_t context = init_and_save_context(ctx, submit_p, SECURITY_INODE_MKNOD, 3 /*argnum*/, 0 /*ret*/);
+
+    u64 *tags = bpf_map_lookup_elem(&params_names_map, &context.eventid);
+    if (!tags) {
+        return -1;
+    }
+
+    struct dentry* dentry = (struct dentry*)PT_REGS_PARM2(ctx);
```

Comment:
Don't we care about the first argument? It tells us the directory on which this dentry is created

Reply:
i think we don't care for this argument because the second argument is the full path - it contains the directory path

Reply:
Can I ask why prefer security_path_mknod to security_inode_mknod?
**security_path_mknod** provide struct path argument which can get absolute file path with mounted file system, while **security_inode_mknod** only provide struct dentry/inode argument which is not aware of mounted file system and can only get path relative to mount point of filesystem.

Reply:
security_path_mknod (and all other path lsm hooks) are enabled by CONFIG_SECURITY_PATH which is not available on all distros: https://elixir.bootlin.com/linux/v5.14.8/source/include/linux/security.h#L1760